game-core/src/main/java/games/strategy/triplea/ResourceLoader.java

```diff
@@ -233,6 +233,29 @@ public boolean hasPath(final String path) {
         .orElse(getMatchingResources(inputPath).stream().findFirst().orElse(null));
   }
 
+  /**
+   * Returns the URL of the resource at the specified path or {@code null} if the resource does not
+   * exist. Tries the given 2 paths in order first in the map resources then engine resources.
+   *
+   * @param inputPath (The name of a resource is a '/'-separated path name that identifies the
+   *     resource. Do not use '\' or File.separator)
+   * @param inputPath2 Same as inputPath but this takes second priority when loading
+   */
+  public @Nullable URL getResource(final String inputPath, final String inputPath2) {
+    final String path = resourceLocationTracker.getMapPrefix() + inputPath;
+    final String path2 = resourceLocationTracker.getMapPrefix() + inputPath2;
+    return getMatchingResources(path).stream()
+        .findFirst()
+        .orElse(
+            getMatchingResources(path2).stream()
+                .findFirst()
+                .orElse(
+                    getMatchingResources(inputPath).stream()
+                        .findFirst()
+                        .orElse(
+                            getMatchingResources(inputPath2).stream().findFirst().orElse(null))));
```

Comment:
This isn't pretty especially with the new google formatting. Open to ideas on refactoring this.

Reply:
I'd approach a refactor by first removing the potentially unused second parameter. The usage of this method would then look like:

```
    return Optional.ofNullable(resourceLoader.getResource(FILE_NAME_BASE + id.getName() + "/" + baseImageName + ".png"))
        .or(resourceLoader.getResource(FILE_NAME_BASE + baseImageName + ".png")
```


For the method itself, the repeated part is:
```
return getMatchingResources(path).stream()
        .findFirst()
```

To fix that, I'd add a helper method, eg:

```
private Optional<URL> findResource(final String searchPath) {
   return getMatchingResources(searchPath)
        .stream()
        .findFirst()
}
```

Then using that helper method, this method becomes:

```
 public @Nullable URL getResource(final String inputPath) {
     return findResource(resourceLocationTracker.getMapPrefix() + inputPath)
                    .or(() ->  findResource(inputPath))
                    .orElse(null);

```

Reply:
Refactored to use a helper method. Though your suggestion is slightly incorrect as it would change the order of the checks. We want to check the folders in this order: map nations, map units, engine nations, engine units.